compiler/next/include/chpl/queries/query-impl.h

```diff
@@ -118,10 +119,19 @@ QueryMapResult<ResultType,ArgTs...>::markUniqueStringsInResult(Context* context)
 template<typename... ArgTs>
 void Context::queryBeginTrace(const char* traceQueryName,
                               const std::tuple<ArgTs...>& tupleOfArg) {
-  if (enableDebugTracing) {
-    printf("QUERY BEGIN     %s (", traceQueryName);
-    queryArgsPrint(tupleOfArg);
-    printf(")\n");
+
+  if (breakSet || enableDebugTracing) {
+    auto args = queryArgsToStrings(tupleOfArg);
+    size_t queryAndArgsHash = hash_combine(hash(traceQueryName), hash(args));
+    if (breakSet && queryAndArgsHash == breakOnHash) {
```

Comment:
Let's put the breakpoint logic after the tracing logic, so if one has a breakpoint and also is doing tracing, they don't have to step through the tracing. The tracing output being first might additionally help indicate where the breakpoint is.